design_docs/mpas_xarray_class.md

```diff
@@ -0,0 +1,185 @@
+<h1> Title: Moving variable mapping outside of mpas_xarray <br>
+Xylar Asay-Davis <br>
+date: 2017/02/10 <br>
+</h1>
+<h2> Summary </h2>
+In discussions with @pwolfram, it became clear that we would like to keep
+mpas_xarray as general as possible, rather than adding code specific to
+MPAS-Analysis.  In particular, the capability for mapping variable names
+that is currently part of mpas_xarray is likely a capability that only
+MPAS-Analysis will need when opening xarray data sets.  Likewise, there is
+a desire for mpax_xarray not to use any of the functionality outside of its
+own module so that it remains autonomous from MPAS-Analysis.
+
+At the same time, it is desirable for efficiency and parallelism to perform
+certain operations during the preprocessing step within xarray, rather than
+constructing a data set first and then (in serial) performing manipulations
+(e.g. creating a time coordinate and slicing variables).  
+
+To solve these problems, this document proposes a solution in which a new 
+MpasXarray class is created with two methods, `open_multifile_dataset` and
+`_preprocess` that can be overloaded by subclasses.  A subclass, 
+MpasAnalysisXarray, is also added in which these methods are overloaded to
+add support for (among other things) mapping variables.
```

Comment:
@pwolfram, my suggested implementation is definitely subclassing, as that's what I've already implemented.  I'm not sure what you mean by class/functional composition.  If you'd like to clarify the pros of that approach, I'm definitely willing to consider an alternative.  I guess the pros would have to be significant for me to want to redo what I've done.